Icebnb/Icebnb/MapViewWidget.swift

```diff
@@ -10,14 +10,17 @@ import UIKit
 import MapKit
 import InstantSearch
 import InstantSearchCore
+import AlgoliaSearch
 
 class MapViewWidget: MKMapView, AlgoliaWidget, ResultingDelegate {
   
   func on(results: SearchResults?, error: Error?, userInfo: [String: Any]) {
     
-    let annotations = self.annotations
-    removeAnnotations(annotations)
-    
+    let searchParams = userInfo["params"] as? SearchParameters
+    if searchParams == nil || searchParams?.page == 0 {
```

Comment:
Probably an opportunity here to add some helper to check if there are search parameters or not. We can do a better developer experience for geo-search related features (we already discussed this live, maybe it's a good idea to add an Asana task